public/app/plugins/datasource/loki/queryUtils.ts

```diff
@@ -76,6 +77,23 @@ export function getHighlighterExpressionsFromQuery(input = ''): string[] {
   return results;
 }
 
+/**
+ * Returns search terms from a LogQL query with variables.
+ * Given a variable name $search with value foo:
+ * E.g., `{} |= $search |= "bar" != "baz"` returns `['foo']`.
+ */
+export function getVariableHighlighterExpressionsFromQuery(expr: string, templateSrv: TemplateSrv) {
+  const searchWords: string[] = [];
+
+  templateSrv.getVariables().forEach((variable) => {
+    if (templateSrv.containsTemplate(expr) && expr.includes(`$${variable.name}`)) {
+      searchWords.push(templateSrv.replace(`$${variable.name}`));
+    }
+  });
+
+  return searchWords;
```

Comment:
This is a bit tricky and not the usual way how to replace variables. 
There is a bunch of ways how users can write variables. E.g. `$var` or `${var}` or `${var:raw} is all valid.

We should use a method like `interpolateVariablesInQueries`, but then we do need access to `scopedVars`. I was thinking, if we should rather use the `response.schema.meta.executedQueryString` property that has the variables already replaced (after string manipulation): 
![image](https://github.com/user-attachments/assets/945a185a-8944-4e9f-a7b2-6c8f4c5daffd)

Reply:
Done. Let me know what you think. Great idea, btw.